tools/cold-backup/run.sh

```diff
@@ -30,109 +30,94 @@ FILE_NAME=""
-      kubectl -n "$NAMESPACE" scale "$zoo_ss" --replicas "$ZK_REPLICAS"
-      wait_for "condition=Ready" "pod -l strimzi.io/name=$CLUSTER_NAME-zookeeper"
-    fi
-    local kafka_ss && kafka_ss="$(kubectl -n "$NAMESPACE" get sts "$CLUSTER_NAME"-kafka -o name --ignore-not-found)"
-    if [[ -n $kafka_ss ]]; then
-      kubectl -n "$NAMESPACE" scale "$kafka_ss" --replicas "$KAFKA_REPLICAS"
-      wait_for "condition=Ready" "pod -l strimzi.io/name=$CLUSTER_NAME-kafka"
-    fi
-    local co_deploy && co_deploy="$(kubectl -n "$NAMESPACE" get deploy strimzi-cluster-operator -o name --ignore-not-found)"
-    if [[ -n $co_deploy ]]; then
-      kubectl -n "$NAMESPACE" scale "$co_deploy" --replicas 1
-      wait_for "condition=Ready" "pod -l strimzi.io/kind=cluster-operator"
-    fi
-    wait_for "condition=Ready" "pod -l strimzi.io/name=$CLUSTER_NAME-kafka"
+error() {
+  echo "$@" 1>&2
+  exit 1
+}
+
+check_kube_conn() {
+  kubectl version -o yaml --request-timeout=5s 1>/dev/null
+}
+
+# wait_for "condition=Ready" "kafkas.kafka.strimzi.io/my-cluster" "my-namespace"
+# wait_for "condition=Ready" "pod -l strimzi.io/kind=cluster-operator" "my-namespace"
+# wait_for "delete" "kafkas.kafka.strimzi.io/my-cluster" "my-namespace"
+wait_for() {
+  local condition="${1-}" resource="${2-}" namespace="${3-}" timeout_sec=300
+  if [[ -z $condition || -z $resource || -z $namespace ]]; then
+    error "Missing parameters"
+  fi
+  echo "Waiting for \"$condition\" on \"res=$resource\" in \"ns=$namespace\""
+  local res_cmd="kubectl -n $namespace get $resource -o name --ignore-not-found"
+  local con_cmd="kubectl -n $namespace wait --for=$condition $resource --timeout=${timeout_sec}s"
+  # waiting for resource creation
+  if [[ $condition != "delete" ]]; then
+    local i=0; while [[ ! $($res_cmd) && $i -lt $timeout_sec ]]; do
+      i=$((i+1)) && sleep 1
+    done
+  fi
+  # waiting for condition
+  $con_cmd
+}
+
+annotate() {
+  local kind="$1" name="$2" annotation="$3" namespace="$4"
+  if [[ -z $kind || -z $name || -z $annotation || -z $namespace ]]; then
+    error "Missing parameters"
+  fi
+  local res && res="$(kubectl -n "$namespace" get "$kind" "$name" -o name --ignore-not-found)"
+  if [[ -n $res ]]; then
+    kubectl -n "$namespace" annotate "$kind" "$name" "$annotation" --overwrite
   fi
 }
 
 stop_cluster() {
   if [[ -n $NAMESPACE && -n $CLUSTER_NAME ]]; then
     echo "Stopping cluster $CLUSTER_NAME"
     local co_deploy && co_deploy="$(kubectl -n "$NAMESPACE" get deploy strimzi-cluster-operator -o name --ignore-not-found)"
-    if [[ -n $co_deploy ]]; then
-      kubectl -n "$NAMESPACE" scale "$co_deploy" --replicas 0
-      wait_for "delete" "pod -l strimzi.io/kind=cluster-operator"
-    else
-      echo "No local operator found, make sure no cluster-wide operator is watching this namespace"
-      confirm
+    if [[ -z $co_deploy ]]; then
+      confirm "No local operator found, make sure no operator is watching this Kafka cluster"
```

Comment:
Does this mean the operator has to run in the same namespace as the cluster? I guess one more reason to use pausing of the resources rather then scaling down the operator.

Reply:
No. It is already pausing the reconciliation :)